test/testspacesmigration/testspacesmigration.cpp

```diff
@@ -0,0 +1,112 @@
+            QString payloadName;
+            if (op == QNetworkAccessManager::PostOperation && request.url().path().endsWith(QLatin1String("migration/spaces"))) {
+                payloadName = QStringLiteral("migrationpayload.json");
+            } else if (op == QNetworkAccessManager::GetOperation && request.url().path().endsWith(QLatin1String("graph/v1.0/me/drives"))) {
+                payloadName = QStringLiteral("drivespayload.json");
+            }
+            if (payloadName.isEmpty()) {
+                return nullptr;
+            } else {
+                QFile f(QStringLiteral(SOURCEDIR "/test/testspacesmigration/%1").arg(payloadName));
+                Q_ASSERT(f.open(QIODevice::ReadOnly));
+                return new FakePayloadReply(op, request, f.readAll(), this);
+            }
+        });
+        auto cap = TestUtils::testCapabilities();
+        cap[QStringLiteral("migration")] = QVariantMap {
+            { QStringLiteral("space_migration"), QVariantMap { { QStringLiteral("enabled"), true }, { QStringLiteral("endpoint"), QStringLiteral("migration/spaces") } } }
+        };
+        fakeFoler.account()->setCapabilities(cap);
+
+
+        const QUrl personalUrl(QStringLiteral("https://ocis.owncloud.test/dav/spaces/1284d238-aa92-42ce-bdc4-0b0000009157$534bb038-6f9d-4093-946f-133be61fa4e7"));
+        QByteArray folder1Uuid;
+        AccountStatePtr newAccountState = AccountState::fromNewAccount(fakeFoler.account());
+        {
+            auto folder1 = addFolder(newAccountState, QStringLiteral("/root"), QStringLiteral("/"));
+            folder1Uuid = folder1->id();
+
+            auto folder2 = addFolder(newAccountState, QStringLiteral("/Documents"), QStringLiteral("/Documents"));
+            auto folder3 = addFolder(newAccountState, QStringLiteral("/eos/a/Alice"), QStringLiteral("/eos/a/Alice"));
+            auto folder4 = addFolder(newAccountState, QStringLiteral("/Shares"), QStringLiteral("/Shares"));
+
+
+            QVERIFY(fakeFoler.account()->capabilities().migration().space_migration.enabled);
+            auto *migr = new SpaceMigration(newAccountState, fakeFoler.account()->capabilities().migration().space_migration.endpoint, this);
+            QSignalSpy spy(migr, &SpaceMigration::finised);
+            migr->start();
+            QVERIFY(spy.wait());
+            migr->deleteLater();
+
+            QCOMPARE(folder1->webDavUrl(), personalUrl);
+            QCOMPARE(folder1->remotePath(), QStringLiteral("/"));
+
+            QCOMPARE(folder2->webDavUrl(), personalUrl);
+            QCOMPARE(folder2->remotePath(), QStringLiteral("/Documents"));
+
+            QCOMPARE(folder3->webDavUrl(), QUrl(QStringLiteral("https://ocis.owncloud.test/dav/spaces/1284d238-aa92-42ce-bdc4-0b0000009157$31f599a1-94e1-4dfb-8bc9-841a57066d05")));
+            QCOMPARE(folder3->remotePath(), QStringLiteral("/"));
+
+            // The shares folder was not migrated
+            // TODO: should shares be disabled by the migration?
```

Comment:
Should in this case the server rather ask to disable the folder instead of sending an error?